scipy/ndimage/tests/test_morphology.py

```diff
@@ -2357,6 +2357,13 @@ def test_binary_input_as_output(function, iterations, brute_force):
     assert_array_equal(expected, data)
 
 
+# XXX: This ends up calling NI_BinaryErosion, which calls NI_InitFilterOffsets.
+# In the latter, footprint_size is computed and set to 0, so malloc(0) is called
+# for offsets.  Then in NI_BinaryErosion, the offsets pointer would be passed to
+# NI_FilterNext2 where it would be adjusted, which would be invalid.
+# Currently, we catch this right away by checking if the size passed to malloc
+# is 0 or not.
+@pytest.mark.xfail(reason="footprint_size equals 0")
```

Comment:
I think the reason this PR has stalled is because we don't have a convincing reproducer, and modifying our source to demonstrate a problem is a bit tricky from an assessment standpoint.

Importantly, can you provide a reliable method for us to build SciPy from source with some kind of compiler/compiler flags that will actually run into a problem that would affect users with a numerical issue or a crash, <ins>without</ins> first modifying our source code?

So:
- checkout latest SciPy `main` branch
- build/compile instructions that will result in a build of SciPy that experiences the low-level problem you mention (i.e., a failure somewhere in our testsuite)